browser/brave_stats/first_run_util.cc

```diff
@@ -12,22 +12,34 @@
 
 namespace brave_stats {
 
-base::Time GetFirstRunTime(PrefService* local_state) {
 #if BUILDFLAG(IS_ANDROID)
+namespace {
+
+bool g_is_first_run = false;
+
+base::Time InitFirstRunTime(PrefService* local_state) {
+  base::Time now = base::Time::Now();
+  local_state->SetTime(kReferralAndroidFirstRunTimestamp, now);
+  g_is_first_run = true;
+  return now;
+}
+
+}  // namespace
+#endif
+
+base::Time GetFirstRunTime([[maybe_unused]] PrefService* local_state) {
+#if BUILDFLAG(IS_ANDROID)
+  CHECK(local_state);
   // Android doesn't use a sentinel to track first run, so we use a
   // preference instead. kReferralAndroidFirstRunTimestamp is used because
   // previously only referrals needed to know the first run value.
   base::Time first_run_timestamp =
       local_state->GetTime(kReferralAndroidFirstRunTimestamp);
   if (first_run_timestamp.is_null()) {
```

Comment:
it is recommended to use `std::optional` instead of checking `is_null()`. See `base/time/time.h`.

Reply:
not applicable as discussed on slack